File: csharp-projects/Leetcode/0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        numberDictionary = {}
        for i in range(len(nums)):
            difference = target - nums[i]
            if difference in numberDictionary:
                return [i, numberDictionary[difference]]
            else:
                numberDictionary[nums[i]] = i
        return []

File: csharp-projects/Leetcode/0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let indexes = [];
    nums.some((num, index) => {
        let match = nums.indexOf(target - num, index + 1);
        if(match != -1) {
            indexes.push(index);
            indexes.push(match);
            return true;
        }
    })
    return indexes;
};